gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -321,6 +322,11 @@ public void close() throws IOException {
     this.workUnitState.setActualHighWatermark(this.nextWatermark);
     if (isInstrumentationEnabled()) {
       this.statsTracker.emitTrackingEvents(getMetricContext(), this.lowWatermark, this.highWatermark, this.nextWatermark);
+    } else {
```

Comment:
Should we move the isInstrumentationEnabled() check inside emitTrackingEventsWithAdditionalTags() and emit tracking events if it is true?

Reply:
I moved the generateTagsForPartitions() call out so that emitTrackingEvents is only for emitting the events.